### ISSUE TITLE
Softmax Loss Layer GPU Training Bug 

### ISSUE THREAD

**Issue by ChrisWC** [BVLC/caffe#2642]
Training is crashing on a line in softmax_loss_layer.cu while training on the gpu. Specifically, it appears to crash when caffe_gpu_asum(...) is called. I think the specific instance of the call is located in SoftmaxWithLossLayer<Dtype>::Forward_gpu(...)

Some output from before it crashed:

I0623 17:22:58.282510  7185 net.cpp:195] data does not need backward computation.
I0623 17:22:58.282513  7185 net.cpp:236] This network produces output loss
I0623 17:22:58.282541  7185 net.cpp:483] Collecting Learning Rate and Weight Decay.
I0623 17:22:58.282552  7185 net.cpp:248] Network initialization done.
I0623 17:22:58.282557  7185 net.cpp:249] Memory required for data: 1321237252
I0623 17:22:58.282712  7185 solver.cpp:42] Solver scaffolding done.
[libprotobuf WARNING google/protobuf/io/coded_stream.cc:505] Reading dangerously large protocol message.  If the message turns out to be larger than 2147483647 bytes, parsing will be halted for security reasons.  To increase the limit (or to disable these warnings), see CodedInputStream::SetTotalBytesLimit() in google/protobuf/io/coded_stream.h.
[libprotobuf WARNING google/protobuf/io/coded_stream.cc:78] The total number of bytes read was 542361868
E0623 17:22:59.554276  7185 net.cpp:508] Forwarding data
E0623 17:22:59.555039  7185 net.cpp:508] Forwarding data_data_0_split
E0623 17:22:59.555065  7185 net.cpp:508] Forwarding label
E0623 17:22:59.555301  7185 net.cpp:508] Forwarding conv1_1
E0623 17:22:59.557685  7185 net.cpp:508] Forwarding relu1_1
E0623 17:22:59.557713  7185 net.cpp:508] Forwarding conv1_2
E0623 17:22:59.577433  7185 net.cpp:508] Forwarding relu1_2
E0623 17:22:59.577476  7185 net.cpp:508] Forwarding pool1
E0623 17:22:59.577716  7185 net.cpp:508] Forwarding conv2_1
E0623 17:22:59.587126  7185 net.cpp:508] Forwarding relu2_1
E0623 17:22:59.587146  7185 net.cpp:508] Forwarding conv2_2
E0623 17:22:59.600601  7185 net.cpp:508] Forwarding relu2_2
E0623 17:22:59.600622  7185 net.cpp:508] Forwarding pool2
E0623 17:22:59.600814  7185 net.cpp:508] Forwarding conv3_1
E0623 17:22:59.607583  7185 net.cpp:508] Forwarding relu3_1
E0623 17:22:59.607602  7185 net.cpp:508] Forwarding conv3_2
E0623 17:22:59.618160  7185 net.cpp:508] Forwarding relu3_2
E0623 17:22:59.618180  7185 net.cpp:508] Forwarding conv3_3
E0623 17:22:59.628716  7185 net.cpp:508] Forwarding relu3_3
E0623 17:22:59.628736  7185 net.cpp:508] Forwarding pool3
E0623 17:22:59.628954  7185 net.cpp:508] Forwarding pool3_pool3_0_split
E0623 17:22:59.628969  7185 net.cpp:508] Forwarding conv4_1
E0623 17:22:59.634605  7185 net.cpp:508] Forwarding relu4_1
E0623 17:22:59.634625  7185 net.cpp:508] Forwarding conv4_2
E0623 17:22:59.644641  7185 net.cpp:508] Forwarding relu4_2
E0623 17:22:59.644661  7185 net.cpp:508] Forwarding conv4_3
E0623 17:22:59.654623  7185 net.cpp:508] Forwarding relu4_3
E0623 17:22:59.654644  7185 net.cpp:508] Forwarding pool4
E0623 17:22:59.654839  7185 net.cpp:508] Forwarding pool4_pool4_0_split
E0623 17:22:59.654855  7185 net.cpp:508] Forwarding conv5_1
E0623 17:22:59.658529  7185 net.cpp:508] Forwarding relu5_1
E0623 17:22:59.658548  7185 net.cpp:508] Forwarding conv5_2
E0623 17:22:59.662082  7185 net.cpp:508] Forwarding relu5_2
E0623 17:22:59.662102  7185 net.cpp:508] Forwarding conv5_3
E0623 17:22:59.665670  7185 net.cpp:508] Forwarding relu5_3
E0623 17:22:59.665690  7185 net.cpp:508] Forwarding pool5
E0623 17:22:59.665900  7185 net.cpp:508] Forwarding fc6
E0623 17:22:59.716789  7185 net.cpp:508] Forwarding relu6
E0623 17:22:59.716830  7185 net.cpp:508] Forwarding drop6
E0623 17:22:59.722168  7185 net.cpp:508] Forwarding fc7
E0623 17:22:59.730558  7185 net.cpp:508] Forwarding relu7
E0623 17:22:59.730597  7185 net.cpp:508] Forwarding drop7
E0623 17:22:59.730711  7185 net.cpp:508] Forwarding score59
E0623 17:22:59.731173  7185 net.cpp:508] Forwarding upscore2
E0623 17:22:59.731371  7185 net.cpp:508] Forwarding upscore2_upscore2_0_split
E0623 17:22:59.731387  7185 net.cpp:508] Forwarding score-pool4
E0623 17:22:59.731726  7185 net.cpp:508] Forwarding crop
E0623 17:22:59.731761  7185 net.cpp:508] Forwarding fuse
E0623 17:22:59.731796  7185 net.cpp:508] Forwarding upsample-fused-16
E0623 17:22:59.732072  7185 net.cpp:508] Forwarding score4_upsample-fused-16_0_split
E0623 17:22:59.732087  7185 net.cpp:508] Forwarding score-pool3
E0623 17:22:59.732421  7185 net.cpp:508] Forwarding crop
E0623 17:22:59.732537  7185 net.cpp:508] Forwarding fuse
E0623 17:22:59.732655  7185 net.cpp:508] Forwarding upsample
E0623 17:22:59.733458  7185 net.cpp:508] Forwarding crop
E0623 17:22:59.733610  7185 net.cpp:508] Forwarding loss

I'm running training on the model at: https://gist.github.com/shelhamer/91eece041c19ff8968ee#file-readme-md

**Comment by pathak22** [comment]
The model that you have listed is not in the caffe-master repo. please ask usage questions on [caffe-users](https://groups.google.com/forum/#!forum/caffe-users) -- this issues tracker is primarily for Caffe development discussion. Thanks!
